src/main/scala/LinterPlugin.scala

```diff
@@ -1944,6 +1945,12 @@ final class LinterPlugin(val global: Global) extends Plugin {
 
             warn(tree, UseLastNotApply(identOrCol(col)))
 
+          case Apply(Select(seq, _apply), List(index))
+            if methodImplements(tree.symbol, SeqLikeApply)
+            && !isLiteral(index)
```

Comment:
The idea behind ignoring `Seq.apply(<literal>)` was to prevent this rule from masking the `UseHeadNotApply` rule for things like `x(1)`.